R/nloptions.R

```diff
@@ -35,36 +35,31 @@
 #'
 #' nl.opts(list(xtol_rel = 1e-8, maxeval = 2000))
 #'
-nl.opts <-
-    function(optlist = NULL)
-    {
-        opts <- list(stopval = -Inf,        # stop minimization at this value
-                     xtol_rel = 1e-6,       # stop on small optimization step
-                 maxeval = 1000,        # stop on this many function evaluations
-                 ftol_rel = 0.0,        # stop on change times function value
-                 ftol_abs = 0.0,        # stop on small change of function value
-                 check_derivatives = FALSE,
-                 algorithm = NULL       # will be filled by each single function
-                )
+nl.opts <- function(optlist = NULL) {
+  opts <- list(
+    stopval = -Inf,            # stop minimization at this value
+    xtol_rel = 1e-6,           # stop on small optimization step
+    maxeval = 1000,            # stop on this many function evaluations
+    ftol_rel = 0.0,            # stop on change times function value
+    ftol_abs = 0.0,            # stop on small change of function value
+    check_derivatives = FALSE,
+    algorithm = NULL           # will be filled by each single function
+  )
 
-    if (is.null(optlist))
-        return(opts)
+  if (is.null(optlist)) return(opts)
 
-    if (!is.list(optlist) || "" %in% names(optlist))
-        stop("Argument 'optlist' must be a list of named (character) objects.")
+  if (!is.list(optlist) || "" %in% names(optlist))
+    stop("Argument 'optlist' must be a list of named (character) objects.")
 
-    namc <- match.arg(names(optlist), choices=names(opts), several.ok=TRUE)
-    if (!all(names(optlist) %in% names(opts)))
-        warning("Unknown names in control: ",
-                names(optlist)[!(names(optlist) %in% names(opts))])
+  for (option_name in names(optlist))
+    opts[[option_name]] <- optlist[[option_name]]
 
-    if (!is.null(namc))
-        opts[namc] <- optlist[namc]
+  if ("algorithm" %in% names(opts)) {
+    cli::cli_alert_warning(
```

Comment:
I don't think you import or even suggests `cli` yet.  On the margin, could it not be base R's `warning()` here as before?

Reply:
My bad, I was fixing it while you were reviewing.
A commit is on its way. Thanks

Reply:
I am though thinking about replacing calls to `warning()`, `stop()` and `cat()` to use more a elegant command-line interface as provided by **cli**. But it's not urgent and purely aesthetics.

Reply:
Sure, your call.  And these days the penny dropped in most places and many of these convenience packages have a low or zero count for _hard_ dependencies and moved the long tail of things they may bring in to Suggests.  `glue` is similarly popular with a narrow hard dependency list.